### PR TITLE
Add stack

### PR DESCRIPTION
**Pull request by prabum3**
Hello Stacks!

**Stack init**:

- Uses `cookiecutter` to generate code from https://github.com/prabum3/farm-stack-template with given inputs
- Sets the webapp to use the given container registry for pulling images
- Sets the MongoDB Connection string as a env variable in the webapp.

**Stack deploy**:

- Builds the backend and nginx images and pushes to the Azure Container Registry of the user.
- Sets the web app to use the docker-compose file using azure cli which should trigger a restart.

### CODE REVIEW

File: .github/workflows/stack-init.yml [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+          docker image build -t $IMAGE_NAME .
+          docker push $IMAGE_NAME
+
+      - name: Build and push docker image (fastapi)
+        run: |
+          cd ./server
+          IMAGE_NAME=${{ secrets.AZURE_CONTAINER_REGISTRY }}/${{ secrets.WEB_APP_NAME }}-fastapi-image:latest
+          echo "building image $IMAGE_NAME"
+          docker image build -t $IMAGE_NAME .
+          docker push $IMAGE_NAME
```
Comment (andyfeller) [inline .github/workflows/stack-init.yml:74]: Curious if you could modify the workflows in the PR to take advantage of reusable workflows:
https://docs.github.com/en/actions/learn-github-actions/reusing-workflows#calling-a-reusable-workflow

File: .github/workflows/build-and-deploy.yml [hunk 2]
```diff
@@ -0,0 +1,55 @@
+name: "Build and deploy farm stack on azure"
```
Comment (andyfeller) [inline .github/workflows/build-and-deploy.yml:1]: Unsure if you want separate workflows for building and deploying, maybe trigger the latter on the creation of a release / tag?

File: README.md [hunk 3]
```diff
@@ -1 +1,51 @@
-### FARM Stack on Azure
\ No newline at end of file
+## FARM Stack on Azure 🧑‍🌾
```
Comment (andyfeller) [inline README.md]:
```suggestion
# FARM Stack on Azure 🧑‍🌾
```

😬  minor nit

File: .github/workflows/stack-init.yml (hunk 1 above)
Comment (prabum3) [reply .github/workflows/stack-init.yml:74]: Merging this PR now, will iterate on having reusable workflows

**Pull request merged**